### PR TITLE
refactor of ruby install and addition of cheetah specific worker recipe

### PR DESCRIPTION
**Pull request by JoeNunnelley**
addition of the cheetah-worker recipe.

### CODE REVIEW

File: recipes/worker-cheetah.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+).each do |pkg|
+  package pkg
+end
+
+directory '/etc/clamav'
+template '/etc/clamav/clamd.conf' do
+  source 'clamd.conf.erb'
+end
+
+gem_package 'selenium-webdriver' do
```
Comment (roarkb) [inline recipes/worker-cheetah.rb]: can't all the gems just get managed via Cheetah's Gemfile/Gemfile.lock?

Comment (JoeNunnelley) [inline recipes/worker-cheetah.rb]: this is pre-provisioning as part of the ami image so we don't have to at run time.

File: attributes/default.rb [hunk 2]
```diff
@@ -1,6 +1,21 @@
 
 # Set the ubuntu user to be a member of the rbenv group
 default['rbenv']['group_users'] = ['ubuntu']
+# Define the rbenv environment
+default['rbenv']['user_installs'] = [
+  { 'user' => "#{default['rbenv']['group_users']}",
```
Comment (michaelb990) [inline attributes/default.rb]: formatting

File: recipes/_ruby.rb [hunk 3]
```diff
@@ -1,20 +1,42 @@
 include_recipe 'rbenv'
-include_recipe 'rbenv::ruby_build'
+include_recipe 'ruby_build'
```
Comment (michaelb990) [inline recipes/_ruby.rb]: why are we changing this?

File: recipes/_ruby.rb [hunk 4]
```diff
@@ -1,20 +1,42 @@
...
+  file '/home/ubuntu/.bashrc' do
+    mode 0755
+    user member
+    action :create_if_missing
+  end
+
+  bash 'add rbenv path' do
+    user member
+    code <<-EOH
+      echo 'export PATH="/opt/rbenv/bin:/opt/rbenv/plugins/ruby_build/bin:$PATH"' >> /home/#{member}/.bashrc
```
Comment (michaelb990) [inline recipes/_ruby.rb]: why is the `ruby_build` part necessary?

File: recipes/_ruby.rb [hunk 5]
```diff
@@ -1,20 +1,42 @@
 include_recipe 'rbenv'
-include_recipe 'rbenv::ruby_build'
+include_recipe 'ruby_build'
+
+members = node['rbenv']['group_users']
+install_details = node['rbenv']['user_installs'][0]
```
Comment (michaelb990) [inline recipes/_ruby.rb]: this is a code smell. if the attributes were organized better, you wouldn't have to do this. what benefit are you getting by storing the rbenv-specific attributes side-by-side with the attributes that you're using in this recipe?

File: recipes/worker-cheetah.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
+user_home = '/home/ubuntu'
+
+include_recipe 'jenkins-workers::_common'
+include_recipe 'jenkins-workers::_clortho'
+include_recipe 'jenkins-workers::_docker-registry-sync'
```
Comment (michaelb990) [inline recipes/worker-cheetah.rb]: is cheetah going to be pushing docker images?

File: attributes/default.rb (hunk 2 above)
Comment (JoeNunnelley) [inline attributes/default.rb]: fixed

File: recipes/worker-cheetah.rb (hunk 6 above)
Comment (JoeNunnelley) [inline recipes/worker-cheetah.rb]: no, removing.

File: recipes/_ruby.rb (hunk 5 above)
Comment (JoeNunnelley) [inline recipes/_ruby.rb]: updated to utilize the attribute structure in the defaults.rb more dynamically and avoid this clunkiness

File: recipes/_ruby.rb (hunk 4 above)
Comment (JoeNunnelley) [inline recipes/_ruby.rb]: its is or we don't get the rbenv install functionality which allows us to manage multiple versions on the node.

File: recipes/_ruby.rb (hunk 3 above)
Comment (JoeNunnelley) [inline recipes/_ruby.rb]: its unnecessary, reverting

File: recipes/_ruby.rb (hunk 4 above)
Comment (michaelb990) [inline recipes/_ruby.rb]: that's not true for my local rbenv or from a quick glance at the ruby build documentation.

Comment (JoeNunnelley) [inline recipes/_ruby.rb]: you appear to be correct. removing.

**Pull request merged**